commons-packet/commons-packet-manager/src/main/java/io/mosip/commons/packet/keeper/PacketKeeper.java

```diff
@@ -1,296 +1,296 @@
+        Manifest manifest = new Manifest();
+
+        Map<String, Object> metaMap = getAdapter().getMetaData(PACKET_MANAGER_ACCOUNT, id, null);
+
+        metaMap.entrySet().forEach(entry -> {
+            Map<String, Object> tempMap = (Map<String, Object>) entry.getValue();
+            PacketInfo packetInfo = PacketManagerHelper.getPacketInfo(tempMap);
+            manifest.getPacketInfos().add(packetInfo);
+        });
+        return manifest;
+    }*/
+
+    /**
+     * Check packet integrity.
+     *
+     * @param packetInfo : the packet information
+     * @return : boolean
+     */
+    public boolean checkIntegrity(PacketInfo packetInfo, byte[] encryptedSubPacket) throws NoSuchAlgorithmException {
+        String hash = CryptoUtil.encodeBase64(HMACUtils2.generateHash(encryptedSubPacket));
+        boolean result = hash.equals(packetInfo.getEncryptedHash());
+        LOGGER.info(PacketManagerLogger.SESSIONID, PacketManagerLogger.REGISTRATIONID,
+                getName(packetInfo.getId(), packetInfo.getPacketName()), "Integrity check : " + result);
+        return result;
+    }
+
+    /**
+     * Check integrity and signature of the packet
+     *
+     *
+     * @param packet
+     * @param encryptedSubPacket
+     * @return
+     */
+    public boolean checkSignature(Packet packet, byte[] encryptedSubPacket) throws NoSuchAlgorithmException {
+        boolean result = disablePacketSignatureVerification ? true :
+        		getCryptoService().verify(packet.getPacketInfo().getRefId(), packet.getPacket()
+        				, CryptoUtil.decodeBase64(packet.getPacketInfo().getSignature()));
+        if (result)
+            result = checkIntegrity(packet.getPacketInfo(), encryptedSubPacket);
+        LOGGER.info(PacketManagerLogger.SESSIONID, PacketManagerLogger.REGISTRATIONID,
+                getName(packet.getPacketInfo().getId(), packet.getPacketInfo().getPacketName()), "Integrity and signature check : " + result);
+        return result;
+    }
+
+    /**
+     * Get packet
+     *
+     * @param packetInfo : packet info
+     * @return : Packet
+     */
+    public Packet getPacket(PacketInfo packetInfo) throws PacketKeeperException {
+        try {
+            InputStream is = getAdapter().getObject(PACKET_MANAGER_ACCOUNT, packetInfo.getId(), packetInfo.getSource(),
+                    packetInfo.getProcess(), getName(packetInfo.getId(), packetInfo.getPacketName()));
+            if (is == null) {
+                LOGGER.error(PacketManagerLogger.SESSIONID, PacketManagerLogger.REGISTRATIONID,
+                        getName(packetInfo.getId(), packetInfo.getPacketName()), packetInfo.getProcess() + " Packet is not present in packet store.");
+                throw new PacketKeeperException(ErrorCode.PACKET_NOT_FOUND.getErrorCode(), ErrorCode.PACKET_NOT_FOUND.getErrorMessage());
+            }
+            byte[] encryptedSubPacket = IOUtils.toByteArray(is);
+            // TODO: need to get refid from metainfo
```

Comment:
I think this should be completed before merge because the decrypt method is already expecting the ref id and decryption will fail